centreon/src/App/MonitoringConfiguration/Infrastructure/ApiPlatform/State/Command/UpdateCommandProcessor.php

```diff
@@ -0,0 +1,82 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace App\MonitoringConfiguration\Infrastructure\ApiPlatform\State\Command;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use App\MonitoringConfiguration\Application\Command\UpdateCommandCommand;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\Command;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\CommandComment;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\CommandId;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\CommandLine;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\CommandName;
+use App\MonitoringConfiguration\Domain\Aggregate\Command\CommandTypeEnum;
+use App\MonitoringConfiguration\Domain\Aggregate\Connector\ConnectorId;
+use App\MonitoringConfiguration\Infrastructure\ApiPlatform\Resource\Command\CommandResource;
+use App\Shared\Application\Command\CommandBus;
+use App\Shared\Infrastructure\Legacy\LegacySecurity;
+use App\Shared\Infrastructure\TransformerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Webmozart\Assert\Assert;
+
+/**
+ * @implements ProcessorInterface<CommandResource, CommandResource>
+ */
+final readonly class UpdateCommandProcessor implements ProcessorInterface
+{
+    /**
+     * @param TransformerInterface<Command, CommandResource> $transformer
+     */
+    public function __construct(
+        private CommandBus $commandBus,
+        #[Autowire(service: ResourceCommandTransformer::class)]
+        private TransformerInterface $transformer,
+        private LegacySecurity $legacySecurity,
+    ) {
+    }
+
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = []): CommandResource
+    {
+        Assert::keyExists($uriVariables, 'id');
+        Assert::integer($uriVariables['id']);
+        $commandId = new CommandId($uriVariables['id']);
+
+        $command = new UpdateCommandCommand(
+            id: $commandId,
+            type: isset($data->type) ? CommandTypeEnum::fromName($data->type) : null,
+            name: isset($data->name) ? new CommandName($data->name) : null,
+            commandLine: isset($data->commandLine) ? new CommandLine($data->commandLine) : null,
+            comment: isset($data->comment) ? new CommandComment($data->comment) : null,
+            isShellEnabled: $data->isShellEnabled ?? null,
+            isActivated: $data->isActivated ?? null,
+            connectorId: isset($data->connector) ? new ConnectorId($data->connector->id) : null,
```

Comment:
```suggestion
            connectorId: isset($data->connector) ? new ConnectorId($data->connectorId) : null,
```
and in the payload use connector_id instead of connector.id

Reply:
well I ve tested this approach and if we use `connectorId` we will need to add a property to `CommandResource`, I think better use the existing `connector->id`